test-app/package.json

```diff
@@ -22,14 +23,45 @@
   ],
   "devDependencies": {
     "@babel/core": "^7.24.4",
-    "@babel/eslint-parser": "^7.24.1",
-    "@babel/plugin-proposal-decorators": "^7.24.1",
     "@ember/optional-features": "^2.1.0",
     "@ember/string": "^3.1.1",
     "@ember/test-helpers": "^3.3.0",
     "@embroider/compat": "^3.4.8",
     "@embroider/core": "^3.4.8",
     "@embroider/webpack": "^4.0.0",
+    "@glint/environment-ember-loose": "^1.4.0",
+    "@glint/template": "^1.4.0",
+    "@tsconfig/ember": "^3.0.8",
+    "@types/ember": "^4.0.11",
```

Comment:
we no longer need to use all these `@types/*` packages, we can use the built in types -- see here as well: 

https://github.com/emberjs/data/blob/main/guides/typescript/0-installation.md